zetaclient/maintenance/shutdown_listener.go

```diff
@@ -55,6 +55,38 @@ func (o *ShutdownListener) Listen(ctx context.Context, action func()) {
 	)
 
 	bg.Work(ctx, o.waitForUpdate, bg.WithName("shutdown_listener.wait_for_update"), withLogger, onComplete)
+	bg.Work(ctx, o.checkIfSyncing, bg.WithName("shutdown_listener.check_if_syncing"), withLogger, onComplete)
+}
+
+// checkIfSyncing checks if the node is syncing ,
+// if it is syncing it returns nil which completes the bg task and calls onComplete
+func (o *ShutdownListener) checkIfSyncing(ctx context.Context) error {
+	isSyncing, err := o.client.GetSyncStatus(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to get sync status initially")
+	}
+	if isSyncing {
+		return nil
+	}
+
+	ticker := time.NewTicker(restartListenerTicker)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ticker.C:
+			isSyncing, err := o.client.GetSyncStatus(ctx)
```

Comment:
this block seems like a duplicate of above, can we just have for loop

Reply:
the begining block returns early , if the node is syncing instead of waitng for the first tick